src/structures/session.ts

```diff
@@ -664,17 +687,24 @@ export default abstract class Session {
             songDuration = 60;
         }
 
-        switch (seekType) {
-            case SeekType.BEGINNING:
-                seekLocation = 0;
-                break;
-            case SeekType.MIDDLE:
-                seekLocation = songDuration * (0.4 + 0.2 * Math.random());
-                break;
-            case SeekType.RANDOM:
-            default:
-                seekLocation = songDuration * (0.6 * Math.random());
-                break;
+        if (!round.songStartedAt) {
```

Comment:
I am confused by this conditional. Are we assuming if its not CLIP mode, songStartedAt is non-null? If so, shouldnt we just check gameType != CLIP?

Reply:
> Are we assuming if its not CLIP mode, songStartedAt is non-null

Not quite. `songStartedAt` is used to track whether the round has been started, and this applies to both clip and non-clip modes. Behavior has not been changed for non-clip mode. However, for clip mode, it is only used to pick the initial starting location, but on a replay/new-clip, the other branches are entered.

To simplify this, I suggest we replace `!round.songStartedAt` with `!clipAction` -- this is the initial entry point to this function for both clip and non-clip game modes.